### PR TITLE
Intra Prediction and the Dim Trait

### PR DESCRIPTION
**Pull request by luctrudeau**

### CODE REVIEW

File: benches/bench.rs [hunk 1]
```diff
@@ -63,7 +63,7 @@ fn native(b: &mut Bencher) {
 
     b.iter(|| {
         for _ in 0..MAX_ITER {
-            pred_dc(&mut o2, 32, &above[..4], &left[..4]);
+        pred_dc::<Block4x4>(&mut o2, 32, &above[..4], &left[..4]);
```
Comment (barrbrain) [inline benches/bench.rs]: I think we just want to drop this bench since the approach it was measuring is deleted.

Comment (lu-zero) [reply benches/bench.rs]: You are right.

File: src/predict.rs [hunk 2]
```diff
@@ -80,10 +80,9 @@ pub fn pred_h<D: Dim>(output: &mut [u16], stride: usize, left: &[u16]) {
   }
 }
 
-pub fn pred_v(output: &mut [u16], stride: usize, above: &[u16], bh: usize) {
-    let bw = above.len();
-    for line in output.chunks_mut(stride).take(bh) {
-        line[..bw].clone_from_slice(above)
+pub fn pred_v<D: Dim>(output: &mut [u16], stride: usize, above: &[u16]) {
+    for line in output.chunks_mut(stride).take(D::H) {
+        line[..D::W].clone_from_slice(above)
```
Comment (lu-zero) [inline src/predict.rs]: `above` would enjoy the same change.

**Comment by lu-zero** [comment]
The next step would be move everything so we have all those in the trait:
``` rust
pub trait Dim {
     const W : usize;
     const H : usize;

     fn pred_dc(...) {
       // here we call Self::W and Self::H
     }
}
```

**Comment by lu-zero** [comment]
https://github.com/rust-av/rav1e/commit/721a24eaf0dc9b110942caeff8119d8292463989 this is what I had in mind.

File: benches/bench.rs (hunk 1 above)
Comment (luctrudeau) [reply benches/bench.rs]: Done

File: src/predict.rs (hunk 2 above)
Comment (luctrudeau) [reply src/predict.rs]: I'm not sure I understand this comment.

Comment (lu-zero) [reply src/predict.rs]: I was thinking about passing `&above[..D::W]`

Comment (luctrudeau) [reply src/predict.rs]: Oh I get it, or else there's a copy operation. Bench seems to indicate a slight speed up from this.

**Comment by lu-zero** [comment]
I'd merge this and see if github autorebase would graciously trim https://github.com/xiph/rav1e/pull/92

**Pull request merged**